File: probes/java/scythe/Probe.java
package scythe;

import java.io.File;

public class Probe {
	
	public static void scythe_probe(String marker) {
		try {
		 String path = System.getenv("SCYTHE_PROBE_DIR");
		 File dir = new File(path);
		 
		 if (!dir.exists() || !dir.isDirectory())
			 return;
		 
		 File file = new File(dir.getPath(), marker + ".scythe_probe");
		 
		 if (file.exists())
			 return;
		 
		 file.setLastModified(System.currentTimeMillis());
		 
		} catch(Exception e) {
		
		}
		 
	}
}

File: VisualStudio2015/Scythe_CSharp_Probe/Probe.cs
using System;
using System.IO;

namespace Scythe
{
    public class Probe
    {
        public static void Scythe_Probe(string marker)
        {
            try
            {
                var folder = Environment.GetEnvironmentVariable("SCYTHE_PROBE_DIR");
                if (!Directory.Exists(folder))
                {
                    if (string.IsNullOrWhiteSpace(folder))
                    {
                        return;
                    }
                    else
                    {
                        Directory.CreateDirectory(folder);
                    }
                }

                var filename = marker + ".scythe_probe";
                var path = Path.Combine(folder, filename);

                if (File.Exists(path))
                {
                    File.SetLastWriteTime(path, DateTime.UtcNow);
                }
                else
                {
                    using (var f = File.Create(path))
                    {
                        // intentionally left blank
                    }
                }
            }
            catch
            {
                // intentionally left blank
            }
        }
    }
}